File: fabric/src/main/net/examplemod/fabric/ExampleModFabric.java
package net.examplemod.fabric;

import net.examplemod.ExampleMod;
import net.fabricmc.api.ClientModInitializer;
import net.fabricmc.api.ModInitializer;

public class ExampleModFabric implements ModInitializer, ClientModInitializer {
    @Override
    public void onInitialize() {
        ExampleMod.INSTANCE.init();
    }

    @Override
    public void onInitializeClient() {

    }
}

File: fabric/src/main/net/examplemod/fabric/ExampleModFabric.kt
package net.examplemod.fabric

import net.examplemod.ExampleMod.init
import net.examplemod.integration.geckolib.fabric.GeckoUtilsImpl
import net.examplemod.registry.Register
import net.fabricmc.api.ClientModInitializer
import net.fabricmc.api.ModInitializer

object ExampleModFabric : ModInitializer, ClientModInitializer {
    override fun onInitialize() = init()

    override fun onInitializeClient() {
        GeckoUtilsImpl.registerAll()
        Register.Client.register()
    }
}